### PR TITLE
mgr/ssh: optionally specify service names

### PR DESCRIPTION
**Pull request by liewegas**
Teuthology needs (or at least wants) this.  I don't think it's something end-users will have much use for.

### CODE REVIEW

File: src/pybind/mgr/orchestrator.py [hunk 1]
```diff
@@ -597,8 +597,18 @@ class PlacementSpec(object):
     """
     def __init__(self, label=None, nodes=[]):
         self.label = label
-        self.nodes = nodes
 
+        def split_host(host):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator.py]: Can we make move this to toplovel?

File: src/pybind/mgr/orchestrator_cli/module.py [hunk 2]
```diff
@@ -551,6 +551,23 @@ def _update_mgrs(self, num, hosts=None):
...
+            a = host.split('=', 1)
+            if len(a) == 1:
+                return (a[0], None)
+            else:
+                assert len(a) == 2
+                return tuple(a)
+
+        if hosts:
+            try:
+                hosts = list(map(split_host, hosts))
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator_cli/module.py]:
```suggestion
                hosts = list(map(orchestrator.split_host, hosts))
```

File: src/pybind/mgr/orchestrator_cli/module.py [hunk 3]
```diff
@@ -570,11 +587,12 @@ def _update_mons(self, num, hosts=None):
         def split_host(host):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator_cli/module.py]: And remove this copy

File: src/pybind/mgr/ssh/module.py [hunk 4]
```diff
@@ -200,18 +200,23 @@ def config_notify(self):
...
-    def get_unique_name(self, existing, prefix=None):
+    def get_unique_name(self, existing, prefix=None, forcename=None):
         """
         Generate a unique random service name
         """
         while True:
-            if prefix:
-                name = prefix + '.'
+            if forcename:
+                name = forcename
```
Comment (sebastian-philipp) [inline src/pybind/mgr/ssh/module.py]: :-1: If we have forcename != None, `get_unique_name` is no longer generating anything at all.

Comment (liewegas) [reply src/pybind/mgr/ssh/module.py]: it's still ensuring the name isn't already taken by another daemon in the cluster

File: src/pybind/mgr/orchestrator_cli/module.py (hunk 3 above)
Comment (liewegas) [reply src/pybind/mgr/orchestrator_cli/module.py]: there are 2 places that use this version, and one place that uses a different version of split_host... wasn't sure if it belonged at top level

File: src/pybind/mgr/ssh/module.py (hunk 4 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/ssh/module.py]: Still `forcename` defeats the outer loop entirely. I don't want to be the guy later to refactor this.

File: src/pybind/mgr/orchestrator_cli/module.py (hunk 3 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator_cli/module.py]: Looks like there is a pattern that this is going to get used more often

File: src/pybind/mgr/ssh/module.py (hunk 4 above)
Comment (liewegas) [reply src/pybind/mgr/ssh/module.py]: udpated!

File: src/pybind/mgr/orchestrator.py [hunk 5]
```diff
@@ -458,13 +492,13 @@ def blink_device_light(self, ident_fault, on, locations):
         """
         raise NotImplementedError()
 
-    def update_mgrs(self, num, hosts):
+    def update_mgrs(self, num, hosts, names):
         # type: (int, List[str]) -> WriteCompletion
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator.py]:
```suggestion
         # type: (int, List[str], List[str]) -> WriteCompletion
```

The parameter `name` needs to be added to all deriving orchestrators.

Comment (liewegas) [reply src/pybind/mgr/orchestrator.py]: oops that hunk doesn't belong, dropped

**Pull request merged**